network/peer.go

```diff
@@ -836,3 +883,53 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// addAlias marks that we have found another
+// IP that we can connect to this peer at.
+//
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.aliasesLock.Lock()
+	defer p.aliasesLock.Unlock()
+
+	p.net.peerAliasIPs[ip.String()] = struct{}{}
+	p.aliases = append(p.aliases, alias{
+		ip:    ip,
+		added: p.net.clock.Time(),
+	})
+}
+
+// releaseAliases removes enables network to
+// reconnect on aliases that have timed out. To
+// force all aliases to be removed, the caller
+// should set now to be the zero time.
+func (p *peer) releaseAliases(now time.Time, holdsStateLock bool) {
+	for {
+		p.aliasesLock.Lock()
```

Comment:
I opted to use granular locking here instead of:
```
next := go func() *alias {
  p.aliasesLock.Lock()
  defer p.aliasesLock.Unlock()

  ....
}()
```

or creating a separate function (i.e. `releaseNextAlias`) but am open to feedback! This approach may be a little more error-prone.